### ISSUE TITLE
Further implement the interface for the player module to interact with new fields

### ISSUE THREAD

**Issue by WPDolan** [uchicago-cs/chiventure#952]
After we understand how other teams intend/desire to interact with the fields in the player module in issue #944, we will implement their desired changes in the `player.h` and `player.c` files of the player module. These new functions will allow other teams to modify a player over the course of a game session as needed by their own modules.

The necessary functions to create/update will be dictated by the documentation created by #944, but will most likely include functions to get/set the existing fields in the `player_t` struct.

**Comment by WPDolan** [comment]
After discussion with the rpg-quests team, @afranzese99 and I have been notified that no functionality currently exists for interacting with quest hashtables. A lack of interface for quest hashes renders us unable to implement our planned functions that would have interacted with the planned `player_quests` field in the player struct. 

For now, all of our additions under the `quests` tab on the player implementation wiki will have to wait until functionality for quest hashes has been implemented. Once the quest team notifies us of their addition, we will implement our planned features.

**Comment by WPDolan** [comment]
@afranzese99 and I have pushed changes to the `action-management/player-interface` branch that implements the desired functionalities from our skilltrees and stats/effects section. 

Functions added include:
- `player_add_combat_skill/player_add_noncombat_skill`, 
-  `player_remove_combat_skill/player_remove_noncombat_skill` , 
- `player_has_combat_skill/player_has_noncombat_skill` 
- `player_add_stat`
- `player_change_stat`
- `player_change_stat_max`
- `player_get_stat_current`
- `player_add_stat_effect`
We have also pushed tests for each of these functions.

We also had to fix a few small bugs in `stats.c` where doubles were accidentally cast to integers. Additionally, we noticed that `get_stat_current` only checks against the corresponding global maximum, while ignoring its local maximum value. From looking at the tests in `test_stats.c` this appears to be an intended design choice, albeit a strange one.

**Comment by WPDolan** [comment]
After discussion with the battles and skilltrees teams, @afranzese99 were informed that the battles team has new ideas for loading skill_moves from the player. We will hold off on implementing planned functionality described under the battles tab until we meet in the future.

**Comment by WPDolan** [comment]
Closing Statement:

The associated [pull request](https://github.com/uchicago-cs/chiventure/pull/1011) to implement the skilltrees and stats/effects components of our interface plan has been merged. 

This merge does not include the quests, battles, and playerclass sections of the [implementation plan](https://github.com/uchicago-cs/chiventure/wiki/Player-~-Interface-Implementation-Plan). We have decided that each of these sections warrant their own individual issues (#1057 and #1058). The battles team also plans to implement their desired functionality for the player module in Issue #1053. These issues will supersede this current issue.
